src/plugins/initiatives/discourse.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+
+import {DomHandler, DomUtils, Parser} from "htmlparser2";
+
+import {type Topic, type Post} from "../discourse/fetch";
+import {type Initiative, type URL} from "./initiative";
+import {topicAddress} from "../discourse/address";
+
+type HeaderToURLsMap = Map<string, $ReadOnlyArray<URL>>;
+
+export async function initiativeFromDiscourseTracker(
+  serverUrl: string,
+  topic: Topic,
+  firstPost: Post
+): Promise<Initiative> {
+  if (firstPost.topicId !== topic.id) {
+    throw new Error("Post is from a different topic");
+  }
+
+  if (firstPost.indexWithinTopic !== 1) {
+    throw new Error("Post is not the first post in the topic");
+  }
+
+  const {title} = topic;
+  const {timestampMs} = firstPost;
+  const tracker = topicAddress(serverUrl, topic.id);
+
+  const map = await groupURLsByHeader(firstPost.cooked);
+  const status = findStatus(map);
+  const champions = singleMatch(map, new RegExp(/^Champions?/i));
+  const contributions = singleMatch(map, new RegExp(/^Contributions?/i));
+  const dependencies = singleMatch(map, new RegExp(/^Dependenc(y|ies)/i));
+  const references = singleMatch(map, new RegExp(/^References?/i));
+
+  const missing = [];
+  if (status === null) missing.push("status");
+  if (!champions) missing.push("champions");
+  if (!dependencies) missing.push("dependencies");
+  if (!references) missing.push("references");
+  if (!contributions) missing.push("contributions");
+
+  if (missing.length > 0) {
+    const missingStr = JSON.stringify(missing);
+    throw new Error(
+      `Missing or malformed headers ${missingStr} for initiative topic "${title}" (${topic.id})`
+    );
+  }
+
+  const completedRE = new RegExp(/^completed?$/i);
+  const completed = status ? completedRE.test(status.trim()) : false;
+
+  return {
+    title,
+    tracker,
+    timestampMs,
+    completed,
+    champions: champions || [],
+    contributions: contributions || [],
+    dependencies: dependencies || [],
+    references: references || [],
+  };
+}
+
+function findStatus(map: HeaderToURLsMap): string | null {
```

Comment:
Testing the value here against the regex makes sense.
That said, unspecified is ambiguous. Because an empty status value is considered incomplete. And the status header missing is considered an error.
Suggesting `boolean | null` return then.

Reply:
Implemented as `boolean | null` with the completion regex moved to this function.